src/app/guards/input/input.guard.ts

```diff
@@ -12,7 +12,7 @@ export interface UnsavedInputCheckingComponent {
  */
 @Injectable()
 export class UnsavedInputGuard
-  implements CanDeactivate<UnsavedInputCheckingComponent>
+  
```

Comment:
I can't tell from the code, but does this canDeactivate guard still work? Worth checking?

Reply:
This same change was made to the `form.guard.ts`, and I can confirm that it still works as expected. I've also verified that the tests are working as expected for the `form.guard.ts` (they would have failed if it no longer worked).

Shockingly, the `input.guard.ts` wasn't even working before the upgrade. The only place this guard is used is on the Harvest Metadata review page, where it'll show a warning if the user tries to navigate away before a mapping is complete.

![image](https://github.com/QutEcoacoustics/workbench-client/assets/33742269/8cf237a9-8fb0-4437-9b52-830e031d4768)

My quick skim understanding is that we've implemented the guard as an interface on the component. When the guard should be applied as a predicate to route navigation.
e.g. here is the `form.guard.ts` implementation.

![image](https://github.com/QutEcoacoustics/workbench-client/assets/33742269/bdef502f-96c1-41f9-95d9-fc48a874a3df)